k2/csrc/cuda/ops.h

```diff
@@ -1,71 +1,149 @@
 // k2/csrc/cuda/ops.h
 
-// Copyright (c)  2020  Xiaomi Corporation (authors: Daniel Povey)
+// Copyright (c)  2020  Xiaomi Corporation (authors: Daniel Povey
+//                                                   Haowen Qiu)
 
 // See ../../LICENSE for clarification regarding multiple authors
 
 #ifndef K2_CSRC_CUDA_OPS_H_
 #define K2_CSRC_CUDA_OPS_H_
 
-#include "k2/csrc/cuda/array.h"
-
+#include <cassert>
+#include <type_traits>
 
-// Note, I'm not sure about the name of this file, they are not ops like in TensorFlow, but
-// procedures..
+#include "k2/csrc/cuda/array.h"
+#include "k2/csrc/cuda/context.h"
+
+// Note, I'm not sure about the name of this file, they are not ops like in
+// TensorFlow, but procedures..
+
+// TODO(haowen): this function may be moved to some header files for log (e.g.
+// `log.h`)
+inline cudaError_t CheckCuda(cudaError_t status) {
```

Comment:
this simple check ignores filename and line number.

Not sure whether [`checkCudaErrors`][1] from cuda samples `helper_cuda.h`
(https://github.com/NVIDIA/cuda-samples/blob/master/Common/helper_cuda.h)
is helpful or not.

[1]: https://github.com/NVIDIA/cuda-samples/blob/master/Common/helper_cuda.h#L595

Reply:
This is just an exercise, definitely we will use some official assert system (meixu is working on it), so will not update in this PR, thanks